### ISSUE TITLE
specs using re-exported Ava

### ISSUE THREAD

**Issue by kosmotaur** [avajs/eslint-plugin-ava#126]
Hi,

For our specs, we put most tools in a test helper, which we import in any given spec like this:

```es6
import {sandbox, test, chance} from '../../helpers/test';
```

Amongst other things, in that helper we import Ava and re-export together with the other goodies:

```es6
import test from 'ava';
// ...
export {sandbox, test, chance};
```

This effectively switched off all the Ava-specific ESLint rules. Is that expected behaviour? Any ideas for workarounds?

**Comment by jfmengels** [comment]
That is expected behavior, in the sense that it is not really feasible using ESLint.
ESLint lints one file at a time and does not provide any context or previous linting info about other files. 

Trying to lint stuff that is imported is like entering a rabbit hole. To make sense out of things, you'd have to go deeper and deeper, maybe even analyze node_modules, which is not something ESLint provides any tools for, so you have to use some kind of hacks to do that.

In our case, you don't even have linting errors when you do something as simple as
```js
import ava from 'ava'; // We only lint when using `test` as the variable name

// -------
import test from 'ava';
test.foo(); // Gets reported by no-unknown-modifiers
const bar = test;
bar.foo(); // Doesn't get reported...
```

which I'll try to work on later, when I have more free time. The reason we don't do that is because ESLint does not provide any helpful tools about the scope (or at least I thought so, but it turns out there are, but they are terribly undocumented), and because it is error-prone (as you might conditionally re-assign variables, which gets very tricky).

I'll close this as we can't really do anything about this unfortunately, but I hope this answers your question :)

**Comment by kosmotaur** [comment]
Thanks, I can see why this would be problematic. I would like to question any level of "smart" here though. This already presents some edge cases (like the one I presented or your `import ava from 'ava';`). What is the benefit of filtering out files to check automatically? Wouldn't it be simpler to both implement and use if the responsibility of correctly scoping the files to check fell on the end user?
This would cover more use cases.
